### ISSUE TITLE
`tf.losses.cosine_distance` still uses `dim`

### ISSUE THREAD

**Issue by lemonzi** [tensorflow/tensorflow#8205]
- Version: current HEAD on master

The loss `tf.losses.cosine_distance` only accepts `dim`, not `axis` as the convention suggests for TF 1.0. It internally passes `axis=(dim,)` to `reduce_sum` though, so there may be a reason to keep the name I'm not getting.

**Comment by prb12** [comment]
@sherrym @josh11b This is a suggested API consistency change.
